40.md

```diff
@@ -0,0 +1,46 @@
+NIP-40
+======
+
+Expiration Timestamp
+-----------------------------------
+
+`draft` `optional` `author:0xtlt`
+
+The `expiration` tag enables users to specify a timestamp at which the message should be considered expired and deleted by relays. This helps ensure that messages are not retained by the network after they are no longer relevant, which helps maintain the privacy of the messages. This is especially useful in cases where a replacement event is not available or feasible.
```

Comment:
I would not justify this nip with privacy. We all know that events once broadcast will get stored for eternity.

```suggestion
The `expiration` tag enables users to specify a timestamp at which the message should be considered expired and deleted by relays. This helps reduce the burden of the network around events with relevance limited in time. This is especially useful in cases where a replacement event is not available or feasible.
```

Reply:
I concur with Giszmo and ok300. I cannot imagine how nostr could create any reasonable expectation of privacy outside of anonymity. Once a post is sent relays can do anything with it, and clients can cache it forever (as mine currently does).

But i ♥ this nip for a different reason: relays will fill up if they don't eventually drop irrelevant events. I've seen relay code that just drops everything after a certain time period (a relay that I cannot use for a blog), and this NIP offers a useful hint to the relay.  I would expire my 'tweets' after a month, and expire my blog posts after 3 years, for example.